src/IssueViz.Security/OpenInIDE/Api/IOpenInIDEStateValidator.cs

```diff
@@ -41,4 +46,103 @@ internal interface IOpenInIDEStateValidator
         /// </remarks>
         bool CanHandleOpenInIDERequest(Uri serverUri, string projectKey, string organizationKey);
     }
+
+    [Export(typeof(IOpenInIDEStateValidator))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class OpenInIdeStateValidator : IOpenInIDEStateValidator
+    {
+        private readonly IInfoBarManager infoBarManager;
+        private readonly IConfigurationProvider configurationProvider;
+        private readonly ILogger logger;
+        private IInfoBar currentInfoBar;
+
+        [ImportingConstructor]
+        public OpenInIdeStateValidator(IInfoBarManager infoBarManager, IConfigurationProvider configurationProvider, ILogger logger)
+        {
+            this.infoBarManager = infoBarManager;
+            this.configurationProvider = configurationProvider;
+            this.logger = logger;
+        }
+
+        public bool CanHandleOpenInIDERequest(Uri serverUri, string projectKey, string organizationKey)
+        {
+            RemoveExistingInfoBar();
+
+            var failureMessage = GetFailureMessage(serverUri, projectKey, organizationKey);
+
+            if (string.IsNullOrEmpty(failureMessage))
+            {
+                return true;
+            }
+
+            AddInfoBar(failureMessage);
+            logger.WriteLine(failureMessage);
+
+            return false;
+        }
+
+        private string GetFailureMessage(Uri serverUri, string projectKey, string organizationKey)
+        {
+            var configuration = configurationProvider.GetConfiguration();
+            string reason = null;
+
+            if (configuration.Mode == SonarLintMode.Standalone)
+            {
+                reason = OpenInIDEResources.Infobar_InvalidStateReason_NotInConnectedMode;
```

Comment:
Is this the message that will be shown if there isn't an open solution, and if so will it make sense to the user?